In [21]:
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
TwitchData = pd.read_csv('Twitch_game_data_edited.csv', encoding = 'unicode_escape')
SteamData = pd.read_csv('SteamCharts_Edited.csv', encoding = 'unicode_escape')

In [3]:
TwitchData

Rank                              Game    Month  Year  Hours_watched  \
0         1                 League of Legends  January  2016       94377226   
1         2  Counter-Strike: Global Offensive  January  2016       47832863   
2         3                            Dota 2  January  2016       45185893   
3         4                       Hearthstone  January  2016       39936159   
4         5       Call of Duty: Black Ops III  January  2016       16153057   
...     ...                               ...      ...   ...            ...   
13195   196                       War Thunder     June  2021         704459   
13196   197                              Muck     June  2021         701456   
13197   198                     Trials Rising     June  2021         698899   
13198   199              Little Nightmares II     June  2021         695130   
13199   200                     Tabletop RPGs     June  2021         688130   

       Peak_viewers  Peak_channels  Avg_viewers  Avg_channels  
0            530270           2903       127021          1833  
1            372654           2197        64378          1117  
2            315083           1100        60815           583  
3            131357            517        53749           317  
4             71639           3620        21740          1549  
...             ...            ...          ...           ...  
13195          8812            223          979           102  
13196         60091            112          975            44  
13197        217333             26          972             6  
13198         43518            105          966            38  
13199         20018            122          957            32  

[13200 rows x 9 columns]

In [4]:
SteamData.head()

Game      Month  Year        Avg     Peak
0  Counter-Strike: Global Offensive  February   2021  741013.24  1123485
1  Counter-Strike: Global Offensive   January   2021  743209.66  1124553
2  Counter-Strike: Global Offensive  December   2020  717803.75  1164396
3  Counter-Strike: Global Offensive  November   2020  668754.58  1037464
4  Counter-Strike: Global Offensive   October   2020  613666.69   943876

## PLAN

1. Only show rows where game is 30,000 or more average (Avg) players in a given month (Steam Data)
2. Only show unique values from step 1
3. Match Twitch stats from Step 2 (names only)
4. Cycle through each with seaborn

5.5k average views on a game should be considered successful. This is because a majority of people don't really watch twitch. This is more of the invested fan domain.

In [5]:
# Making a separate variable where the games have an average of 30,000 players
SuccessGames = SteamData.loc[SteamData['Avg'] > 30000]

In [6]:
SuccessGames

Game       Month  Year        Avg     Peak
0      Counter-Strike: Global Offensive   February   2021  741013.24  1123485
1      Counter-Strike: Global Offensive    January   2021  743209.66  1124553
2      Counter-Strike: Global Offensive   December   2020  717803.75  1164396
3      Counter-Strike: Global Offensive   November   2020  668754.58  1037464
4      Counter-Strike: Global Offensive    October   2020  613666.69   943876
...                                 ...         ...   ...        ...      ...
26046             Monster Hunter: World  September   2018   88575.17   227644
26047             Monster Hunter: World     August   2018  188291.24   329333
26279                      No Man's Sky     August   2016   36976.41   212321
30672                   Resident Evil 2    January   2019   40205.70    74024
31966        Sekiro™: Shadows Die Twice      March   2019   68631.95   124334

[1011 rows x 5 columns]

In [7]:
# I drop the duplicates in order to get a scale of how many games there are on the steam database.
SuccessGames.drop_duplicates(subset=['Game'])

Game      Month  Year        Avg     Peak
0      Counter-Strike: Global Offensive  February   2021  741013.24  1123485
104                              Dota 2  February   2021  404832.13   651615
208       PLAYERUNKNOWN'S BATTLEGROUNDS  February   2021  198957.52   447390
256                        Apex Legends  February   2021  120982.64   196799
260                                Rust  February   2021  117742.27   224276
...                                 ...        ...   ...        ...      ...
25817                  Z1 Battle Royale   October   2017   31263.39    63253
26018             Monster Hunter: World   January   2021   30795.24    56758
26279                      No Man's Sky    August   2016   36976.41   212321
30672                   Resident Evil 2   January   2019   40205.70    74024
31966        Sekiro™: Shadows Die Twice     March   2019   68631.95   124334

[64 rows x 5 columns]

Okay so there are 31966 unique values in this chart. I hope there's a significant amount of losses I have to make for this

In [8]:
# I make another variable of games that hit more than 5500 average monthly viewers on Twitch.
SuccessTwitch = TwitchData.loc[TwitchData['Avg_viewers'] > 5500]

In [9]:
# I drop the duplicate game names to get an idea of how many games there would be on the Twitch side
SuccessTwitch.drop_duplicates(subset=['Game'])

Rank                              Game    Month  Year  Hours_watched  \
0         1                 League of Legends  January  2016       94377226   
1         2  Counter-Strike: Global Offensive  January  2016       47832863   
2         3                            Dota 2  January  2016       45185893   
3         4                       Hearthstone  January  2016       39936159   
4         5       Call of Duty: Black Ops III  January  2016       16153057   
...     ...                               ...      ...   ...            ...   
13020    21               PUBG: BATTLEGROUNDS     June  2021       14126661   
13032    33  Pok<U+00E9>mon FireRed/LeafGreen     June  2021        7717964   
13038    39               Guilty Gear: Strive     June  2021        6259105   
13045    46       Ratchet & Clank: Rift Apart     June  2021        4702938   
13052    53                         StarCraft     June  2021        4370426   

       Peak_viewers  Peak_channels  Avg_viewers  Avg_channels  
0            530270           2903       127021          1833  
1            372654           2197        64378          1117  
2            315083           1100        60815           583  
3            131357            517        53749           317  
4             71639           3620        21740          1549  
...             ...            ...          ...           ...  
13020        114182           1173        19647           731  
13032        225790            125        10734            43  
13038         39623           1081         8705           226  
13045        135505           1592         6540           190  
13052         50524             74         6078            30  

[245 rows x 9 columns]

Now that I have an idea about how many games there are for each chart, along with the values that come with that like average viewers, I can get the list of unique games from each dataframe and combine them into one list~

In [10]:
TwitchNames = SuccessTwitch['Game']
TwitchNames = pd.unique(TwitchNames)

In [11]:
TwitchNames

array(['League of Legends', 'Counter-Strike: Global Offensive', 'Dota 2',
       'Hearthstone', 'Call of Duty: Black Ops III', 'Minecraft',
       'World of Warcraft', 'Z1: Battle Royale', 'Talk Shows & Podcasts',
       'FIFA 16', 'StarCraft II', 'Diablo III', 'Destiny', 'Blade & Soul',
       'World of Tanks', "Tom Clancy's Rainbow Six Siege", 'RuneScape',
       "Tom Clancy's The Division", 'XCOM 2', 'Overwatch',
       'Street Fighter V', 'Stardew Valley', 'Dark Souls III',
       'The Culling', 'Creative', 'Music', 'Heroes of the Storm',
       "Uncharted 4: A Thief's End", 'Poker', 'DOOM', 'DayZ', 'E3 2016',
       'Dead by Daylight', 'Grand Theft Auto V', 'NBA 2K16',
       'Pok<U+00E9>mon GO', 'Arma 3', "No Man's Sky", 'FIFA 17',
       'Battlefield 1', 'Call of Duty: Infinite Warfare',
       'Call of Duty: Modern Warfare Remastered',
       'Pok<U+00E9>mon Sun/Moon', 'Path of Exile', 'Final Fantasy XV',
       'Friday the 13th: The Game', 'Resident Evil 7 biohazard', 'SMITE',

In [12]:
SteamNames = SuccessGames['Game']
SteamNames = pd.unique(SteamNames)

In [13]:
SteamNames

array(['Counter-Strike: Global Offensive', 'Dota 2',
       "PLAYERUNKNOWN'S BATTLEGROUNDS", 'Apex Legends', 'Rust',
       'Team Fortress 2', 'Grand Theft Auto V',
       "Tom Clancy's Rainbow Six Siege", 'Rocket League', 'Path of Exile',
       'Football Manager 2021', 'ARK: Survival Evolved', 'Destiny 2',
       "Sid Meier's Civilization VI", 'PAYDAY 2', 'Wallpaper Engine',
       'Warframe', 'Stardew Valley', "Garry's Mod", 'Dead by Daylight',
       'Dyson Sphere Program', 'Cyberpunk 2077',
       'Total War: WARHAMMER II', 'Unturned', 'Football Manager 2020',
       "Sid Meier's Civilization V", 'Counter-Strike',
       'DARK SOULS\x99 III', 'Fallout 4', 'Fall Guys: Ultimate Knockout',
       'Phasmophobia', 'Mount & Blade II: Bannerlord',
       'Total War: THREE KINGDOMS', 'Crusader Kings III',
       'The Elder Scrolls V: Skyrim',
       'Total War: ROME II - Emperor Edition', 'Warhammer: Vermintide 2',
       'Borderlands 3', 'Dota Underlords', 'Clicker Heroes', 'Starbound',


In [14]:
AllNames = np.concatenate((SteamNames, TwitchNames))

In [15]:
AnalysisGames = SuccessGames.merge(SuccessTwitch, on = "Game").drop_duplicates(subset=['Game'])

In [16]:
AnalysisGames['Game']
#This is the important list of games I'm going to analyze.

0        Counter-Strike: Global Offensive
5742                               Dota 2
12606       PLAYERUNKNOWN'S BATTLEGROUNDS
15003                        Apex Legends
15119                                Rust
15749                  Grand Theft Auto V
19475      Tom Clancy's Rainbow Six Siege
21763                       Rocket League
23583                       Path of Exile
24248               ARK: Survival Evolved
24583                           Destiny 2
25229                            Warframe
25541                      Stardew Valley
25547                    Dead by Daylight
26183                      Cyberpunk 2077
26185        Fall Guys: Ultimate Knockout
26201                        Phasmophobia
26215        Mount & Blade II: Bannerlord
26219                  Crusader Kings III
26220             Warhammer: Vermintide 2
26221                       Borderlands 3
26222                     Dota Underlords
26228                              XCOM 2
26229                        DOOM 

Now that I have the names of the games I'm gonna use. I'm going to check each of these one by one and see how streaming has affected the player count of the game.

## How I'm going to do it
I'm going to go into both csv files on excel, extract all the information of the game, and only have the months/years both datasets have.
The reason I'm doing this is because there aren't as many games as I thought there was.

According to the concerns I had at the beginning of the project, I'm not going to include games that have a competitive scene behind them. This means...
* Counter-Strike: Global Offensive
* Dota 2
* Tom Clancy's Rainbow Six Siege
* Rocket League

Will not be included in this analysis

* Crusader Kings III
* Warhammer: Vermintide 2
* XCOM 2
* Resident Evil 2
* Wolsen: Lords of Mayhem
* Mount and Blade

Didn't have enough data to support a full analysis

## After looking through the data, I think it's best to present the project on 2 criteria.
* Success or Failure
* Multiplayer or Singleplayer

### Multiplayer Games
* PLAYERUNKNOWN'S BATTLEGROUNDS +
* Apex Legends +
* Rust +
* Grand Theft Auto 5 +
* Path Of Exile +
* Destiny 2 +
* Fall Guys: Ultimate Knockout +
* Phasmophobia -
* Realm Royale -
* Dota Underlords -
* Warframe ~
* DeadByDaylight +

### Singleplayer Games
* Ark Survival Evolved
* Stardew Valley
* Borderlands 3
* Tom Clancy's The division
* Monster Hunter World
* No Man's Sky
* Cyberpunk 2077
* Doom Eternal
* Terraria
* Witcher 3